virtualcluster/pkg/syncer/conversion/mutate.go

```diff
@@ -172,7 +173,10 @@ func PodMutateDefault(vPod *v1.Pod, saSecretMap map[string]string, services []*v
 		if err != nil {
 			return err
 		}
-		mutateDNSConfig(p, vPod, vc.Spec.ClusterDomain, nameServer)
+		allowTenantDNSPolicy := featuregate.DefaultFeatureGate.Enabled(featuregate.TenantAllowDNSPolicy)
+		if _, set := vPod.GetLabels()[constants.TenantDisableDNSPolicyMutation]; !allowTenantDNSPolicy && !set {
```

Comment:
I think it should be `!(allowTenantDNSPolicy && set)` ?

Reply:
Isn't this logic the same? `&&` covers that, unless you mean stylistically.

Reply:
The mutate call is skipped if someone sets the constants.TenantDisableDNSPolicyMutation label even if the feature gate is disabled.

Reply:
Oh duh 🤦